clients/java/src/main/java/io/camunda/client/api/search/UserTaskFilter.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright © 2017 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.client.api.search;
+
+import io.camunda.client.api.search.TypedSearchQueryRequest.SearchRequestFilter;
+import io.camunda.client.protocol.rest.DateFilter;
+
+/** Interface for defining user task filters in search queries. */
+public interface UserTaskFilter extends SearchRequestFilter {
+
+  /**
+   * Filters user tasks by the specified key.
+   *
+   * @param value the key of the user task
+   * @return the updated filter
+   */
+  UserTaskFilter userTaskKey(final Long value);
+
+  /**
+   * Filters user tasks by the specified state.
+   *
+   * @param state the state of the user task
+   * @return the updated filter
+   */
+  UserTaskFilter userTaskState(final String state);
```

Comment:
🔧 Do we need the `userTask` prefix on all those filter options? I guess the context "userTask" is clear already here and thus we can remove it. Makes it less verbose. What do you think?

Reply:
I have renamed to only state.